tests/unit/s2n_extension_list_process_test.c

```diff
@@ -276,7 +278,7 @@ int main()
             EXPECT_SUCCESS(s2n_connection_free(conn));
         }
 
-        /* Fails on unexpected parsed_extension */
+        /* Skips on unexpected parsed_extension */
```

Comment:
How does this relate to the RFC? Are we relaxing our expectations to support the insanity sent to us by certain sites? Or is this totally acceptable?

Reply:
The RFC wants us to alert: https://tools.ietf.org/html/rfc8446#section-4.2
> If an implementation receives an extension
   which it recognizes and which is not specified for the message in
   which it appears, it MUST abort the handshake with an
   "illegal_parameter" alert.

We're still not accepting or processing these unexpected extensions, just silently ignoring them instead of ending the connection.

Reply:
Can you add a comment here quoting the RFC text and explaining why we're not doing what it suggests?